contract/contracts/Voting/Types.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+library Types {
+      struct DirectiveVotes {
+        uint256 yes;
+        uint256 no;
+        uint256 abstain;
+    }
+
+    struct Candidate {
+        string name;
+        address candidateAddress;
+        uint256 votes;
+    }
+    struct Member{
+        string name;
+        bool isEligible;
+        bool isVoted;
+        address memberAddress;
+    }   
+    struct Proposal{
```

Comment:
Can structs take on interfaces in solidity? That is can

'interface Proposal {}'
'struct Directive {} is Proposal'
'struct Election {} is Proposal'
So that a function could call
function addProposal(Types.Proposal calldata _proposal) public onlyOwner whenNotPaused {
and then the user could pass in either a Directive or Election?

Reply:
Ah no, turns out interfaces can only be used to define function prototypes that are implemented later on.